src/vuestic-theme/vuestic-components/vuestic-checkbox/VuesticCheckbox.vue

```diff
@@ -27,65 +49,171 @@ export default {
   name: 'vuestic-checkbox',
   props: {
     label: String,
+    value: {
+      type: Boolean,
+      required: true
+    },
     id: {
       type: String,
       default () {
         // We require unique id to show label
         return 'label-' + generateRandomId()
       }
     },
-    checked: {
+    disabled: {
       type: Boolean,
       default: false
     },
-    disabled: {
+    readonly: {
       type: Boolean,
       default: false
     },
-    name: {
+    errorMessage: {
       type: String,
-      default: ''
+      default: null
     },
-    isCircle: {
+    error: {
       type: Boolean,
       default: false
-    },
-    brandColor: {
-      type: String,
-      default: 'primary',
-      validator: value => {
-        return ['primary', 'secondary', 'success', 'info', 'warning', 'danger'].indexOf(value) >= 0
-      }
     }
   },
-  model: {
-    prop: 'checked',
-    event: 'change'
-  },
-  mounted () {
-    this.$emit('change', this.checked)
+  data () {
+    return {
+      onFocus: false
+    }
   },
-  methods: {
-    onChange (event) {
-      this.$emit('change', event.target.checked)
+  computed: {
+    selected: {
+      set (selected) {
+        if (!this.readonly && !this.disabled) {
+          this.$emit('input', selected)
+        }
+      },
+      get () {
+        return this.value
+      }
+    },
+    errorComputed () {
+      // We make error active, if the error-message is not empty and checkbox is not disabled
+      if (!this.disabled) {
+        if (this.errorMessage !== null || this.error) {
```

Comment:
Why not showing error message for disabled checkbox?
Also you check error message for null only, I'd say for any falsy value message should not be shown

I'd simply:

```
showError () {
  return this.errorMessage || this.error
}
```